File: web/lib/src/logic/math/geometry/area.dart
import 'package:calculate_engine/src/logic/math/constant/pi.dart';

// Area of Objects

// Area of a Circle
double circle_area(double radius) {
  return pi() * (radius * radius);
}

// Area of a Cube
double cube_area(double length, double width) {
  return 6 * (length * width);
}

// Area of a Cylinder
double cylinder_area(double height, double radius) {
  return (2 * height * pi() * radius) + (2 * pi() * radius * radius);
}

// Area of a Hemisphere
double hemisphere_area(double radius) {
  return 3 * (pi() * radius * radius);
}

// Area of a Parallelogram
double parallelogram_area(double base, double height) {
  return base * height;
}

// Area of a Rectangle
double rectangle_area(double length, double width) {
  return length * width;
}

// Area of a Sphere
double sphere_area(double radius) {
  return 4 * (pi() * radius * radius);
}

// Area of a Square
double square_area(double length, double width) {
  return length * width;
}

// Area of a Trapezium
double trapezium_area(double baseOne, double baseTwo, double height) {
  return (1 / 2) * (baseOne + baseTwo) * height;
}

// Area of a Triangle
double triangle_area(double base, double height) {
  return (1 / 2) * base * height;
}

File: lib/src/math/geometry/area.zig
const pi_value = @import("/lib/src/math/constant/pi.zig");

// Area of Objects

// Area of a Circle
pub fn circleArea(radius: f64) f64 {
    return pi_value.pi() * (radius * radius);
}

// Area of a Cube
pub fn cubeArea(length: f64, width: f64) f64 {
    return 6 * (length * width);
}

// Area of a Cylinder
pub fn cylinderArea(height: f64, radius: f64) f64 {
    return (2 * height * pi_value.pi() * radius) + (2 * pi_value.pi() * radius * radius);
}

// Area of a Hemisphere
pub fn hemisphereArea(radius: f64) f64 {
    return 3 * (pi_value.pi() * radius * radius);
}

// Area of a Parallelogram
pub fn parallelogramArea(base: f64, height: f64) f64 {
    return base * height;
}

// Area of a Rectangle
pub fn rectangleArea(length: f64, width: f64) f64 {
    return length * width;
}

// Area of a Sphere
pub fn sphereArea(radius: f64) f64 {
    return 4 * (pi_value.pi() * radius * radius);
}

// Area of a Square
pub fn squareArea(length: f64, width: f64) f64 {
    return length * width;
}

// Area of a Trapezium
pub fn trapeziumArea(base_one: f64, base_two: f64, height: f64) f64 {
    return (1 / 2) * (base_one + base_two) * height;
}

// Area of a Triangle
pub fn triangleArea(base: f64, height: f64) f64 {
    return (1 / 2) * base * height;
}
